Wrong behaviour opening file that contains colon : in filename

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When opening a file has `:` in its filename, zed opens a new buffer instead of the file.
For example, when opening a file named `a:b.txt` it opens a new buffer named `a`.
This hapens on lastest stable v0.143.6 and master, self compiled or relaese binary.



### Environment

OS: Gentoo Linux 2.15 x86_64
CPU: AMD Ryzen 7 3700X
GPU: AMD Radeon RX 6500 XT
DE: KDE Plasma 6.1.2
WM: KWin (Wayland)

### If applicable, add mockups / screenshots to help explain present your vision of the feature

relaese binary https://github.com/zed-industries/zed/releases/tag/v0.143.6
![image](https://github.com/zed-industries/zed/assets/19249377/f8b0e295-48b6-4fa8-8281-d78a257b03b8)

compile from master
![image](https://github.com/zed-industries/zed/assets/19249377/d9d156ce-a52f-4432-ad94-4f8180f649f0)


### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-07-11T04:29:56+08:00 [INFO] ========== starting zed ==========
2024-07-11T04:29:56+08:00 [INFO] Opening main db
2024-07-11T04:29:56+08:00 [INFO] perform;
2024-07-11T04:29:56+08:00 [INFO] read_command;
2024-07-11T04:29:56+08:00 [INFO] read_command;
2024-07-11T04:29:56+08:00 [INFO] socket reader;
2024-07-11T04:29:56+08:00 [INFO] new;
2024-07-11T04:29:56+08:00 [INFO] keep_updated;
2024-07-11T04:29:56+08:00 [INFO] Using git binary path: None
2024-07-11T04:29:57+08:00 [INFO] extensions updated. loading 2, reloading 0, unloading 0
2024-07-11T04:29:57+08:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-07-11T04:29:57+08:00 [INFO] perform;
2024-07-11T04:29:57+08:00 [INFO] read_command;
2024-07-11T04:29:57+08:00 [INFO] Opening main db
2024-07-11T04:29:57+08:00 [INFO] read_command;
2024-07-11T04:29:57+08:00 [INFO] socket reader;
2024-07-11T04:29:57+08:00 [INFO] new;
2024-07-11T04:29:57+08:00 [INFO] keep_updated;
2024-07-11T04:29:57+08:00 [INFO] Enabling Vulkan Portability
2024-07-11T04:29:57+08:00 [INFO] Enabling color space support
2024-07-11T04:29:57+08:00 [INFO] set environment variables from shell:/bin/zsh, path:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/bin:/usr/lib/llvm/18/bin:/etc/eselect/wine/bin
2024-07-11T04:29:57+08:00 [INFO] Adapter "AMD Radeon RX 6500 XT (RADV NAVI24)"
2024-07-11T04:29:57+08:00 [INFO] Ray tracing is supported
2024-07-11T04:29:57+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:29:57+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:29:57+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T04:29:57+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T04:29:57+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-07-11T04:29:57+08:00 [INFO] set status on client 0: Authenticating
2024-07-11T04:29:57+08:00 [INFO] Opening main db
2024-07-11T04:29:57+08:00 [INFO] perform;
2024-07-11T04:29:57+08:00 [INFO] read_command;
2024-07-11T04:29:57+08:00 [INFO] read_command;
2024-07-11T04:29:57+08:00 [INFO] socket reader;
2024-07-11T04:29:57+08:00 [ERROR] failed to canonicalize root path: No such file or directory (os error 2)
2024-07-11T04:29:57+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:29:57+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:29:57+08:00 [INFO] new;
2024-07-11T04:29:57+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-07-11T04:29:57+08:00 [INFO] keep_updated;
2024-07-11T04:29:57+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:29:57+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:29:57+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-07-11T04:29:57+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:29:57+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:29:57+08:00 [INFO] set status on client 177024: Connecting
2024-07-11T04:29:57+08:00 [INFO] Opening main db
2024-07-11T04:29:57+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:29:57+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:29:58+08:00 [INFO] Node runtime install_if_needed
2024-07-11T04:29:58+08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-07-11T04:29:59+08:00 [INFO] starting language server. binary path: "/home/skye/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/", args: ["/home/skye/.local/share/zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-07-11T04:29:59+08:00 [INFO] add connection to peer
2024-07-11T04:29:59+08:00 [INFO] add_connection;
2024-07-11T04:29:59+08:00 [INFO] waiting for server hello
2024-07-11T04:29:59+08:00 [INFO] got server hello
2024-07-11T04:29:59+08:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 451, id: 5449048 })
2024-07-11T04:29:59+08:00 [INFO] set status on client 177024: Connected { peer_id: PeerId { owner_id: 451, id: 5449048 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-07-11T04:29:59+08:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-07-10T20:29:59.754Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-07-10T20:29:59.754Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-07-11T04:29:59+08:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "6d232666-e706-4ad6-a15b-0f6856d51ad6",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-07-11T04:29:59+08:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-07-10T20:29:59.773Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-07-10T20:29:59.773Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-07-11T04:31:56+08:00 [ERROR] window not found
2024-07-11T04:31:56+08:00 [ERROR] window not found
2024-07-11T04:32:08+08:00 [INFO] ========== starting zed ==========
2024-07-11T04:32:08+08:00 [INFO] Opening main db
2024-07-11T04:32:08+08:00 [INFO] perform;
2024-07-11T04:32:08+08:00 [INFO] read_command;
2024-07-11T04:32:08+08:00 [INFO] read_command;
2024-07-11T04:32:08+08:00 [INFO] socket reader;
2024-07-11T04:32:08+08:00 [INFO] new;
2024-07-11T04:32:08+08:00 [INFO] Using git binary path: None
2024-07-11T04:32:08+08:00 [INFO] keep_updated;
2024-07-11T04:32:08+08:00 [INFO] extensions updated. loading 2, reloading 0, unloading 0
2024-07-11T04:32:08+08:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-07-11T04:32:08+08:00 [INFO] perform;
2024-07-11T04:32:08+08:00 [INFO] read_command;
2024-07-11T04:32:08+08:00 [INFO] read_command;
2024-07-11T04:32:08+08:00 [INFO] socket reader;
2024-07-11T04:32:08+08:00 [INFO] new;
2024-07-11T04:32:08+08:00 [INFO] keep_updated;
2024-07-11T04:32:08+08:00 [INFO] set status on client 0: Authenticating
2024-07-11T04:32:08+08:00 [INFO] Opening main db
2024-07-11T04:32:08+08:00 [INFO] perform;
2024-07-11T04:32:08+08:00 [INFO] read_command;
2024-07-11T04:32:08+08:00 [INFO] read_command;
2024-07-11T04:32:08+08:00 [INFO] socket reader;
2024-07-11T04:32:08+08:00 [INFO] set environment variables from shell:/bin/zsh, path:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/bin:/usr/lib/llvm/18/bin:/etc/eselect/wine/bin
2024-07-11T04:32:08+08:00 [INFO] Enabling Vulkan Portability
2024-07-11T04:32:08+08:00 [INFO] Enabling color space support
2024-07-11T04:32:08+08:00 [INFO] new;
2024-07-11T04:32:08+08:00 [INFO] keep_updated;
2024-07-11T04:32:08+08:00 [INFO] Adapter "AMD Radeon RX 6500 XT (RADV NAVI24)"
2024-07-11T04:32:08+08:00 [INFO] Ray tracing is supported
2024-07-11T04:32:08+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:32:08+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:32:08+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T04:32:08+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T04:32:08+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-07-11T04:32:08+08:00 [INFO] set status on client 177024: Connecting
2024-07-11T04:32:08+08:00 [INFO] Opening main db
2024-07-11T04:32:08+08:00 [ERROR] failed to canonicalize root path: No such file or directory (os error 2)
2024-07-11T04:32:08+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:32:08+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:32:08+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-07-11T04:32:08+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:32:08+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:32:08+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-07-11T04:32:08+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:32:08+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:32:08+08:00 [INFO] Opening main db
2024-07-11T04:32:08+08:00 [INFO] Using surface present mode MAILBOX
2024-07-11T04:32:08+08:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T04:32:09+08:00 [INFO] Node runtime install_if_needed
2024-07-11T04:32:10+08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-07-11T04:32:10+08:00 [INFO] starting language server. binary path: "/home/skye/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/", args: ["/home/skye/.local/share/zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-07-11T04:32:11+08:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-07-10T20:32:11.636Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-07-10T20:32:11.636Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-07-11T04:32:11+08:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "6d4f1ca8-614d-432f-b11e-fcec43908aba",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-07-11T04:32:11+08:00 [INFO] add connection to peer
2024-07-11T04:32:11+08:00 [INFO] add_connection;
2024-07-11T04:32:11+08:00 [INFO] waiting for server hello
2024-07-11T04:32:11+08:00 [INFO] got server hello
2024-07-11T04:32:11+08:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 451, id: 5450237 })
2024-07-11T04:32:11+08:00 [INFO] set status on client 177024: Connected { peer_id: PeerId { owner_id: 451, id: 5450237 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-07-11T04:32:11+08:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-07-10T20:32:11.674Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-07-10T20:32:11.674Z]",
  "extra": [
    "Telemetry initialized"
  ]
}


<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/util/src/paths.rs
+++ b/crates/util/src/paths.rs
@@ -137,9 +137,78 @@ impl PathWithPosition {
             column: None,
         }
     }
+
     /// Parses a string that possibly has `:row:column` or `(row, column)` suffix.
     /// Ignores trailing `:`s, so `test.rs:22:` is parsed as `test.rs:22`.
     /// If the suffix parsing fails, the whole string is parsed as a path.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use util::paths::PathWithPosition;
+    /// # use std::path::PathBuf;
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: None,
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: Some(1),
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1:2"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: Some(1),
+    ///     column: Some(2),
+    /// });
+    /// ```
+    ///
+    /// # Expected parsing results when encounter ill-formated inputs.
+    /// ```
+    /// # use util::paths::PathWithPosition;
+    /// # use std::path::PathBuf;
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs:a"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: None,
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs:a:b"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: None,
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs::"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: None,
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs::1"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: None,
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1::"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: Some(1),
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs::1:2"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: None,
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1::2"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: Some(1),
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1:2:3"), PathWithPosition {
+    ///     path: PathBuf::from("test_file.rs"),
+    ///     row: Some(1),
+    ///     column: Some(2),
+    /// });
+    /// ```
     pub fn parse_str(s: &str) -> Self {
         let trimmed = s.trim();
         let path = Path::new(trimmed);
@@ -350,69 +419,6 @@ mod tests {
                 (Path::new("test_dirs/1.46/bar_2"), true),
             ]
         );
-    }
-
-    #[test]
-    fn path_with_position_parsing_positive() {
-        let input_and_expected = [
-            (
-                "test_file.rs",
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row: None,
-                    column: None,
-                },
-            ),
-            (
-                "test_file.rs:1",
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row: Some(1),
-                    column: None,
-                },
-            ),
-            (
-                "test_file.rs:1:2",
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row: Some(1),
-                    column: Some(2),
-                },
-            ),
-        ];
-
-        for (input, expected) in input_and_expected {
-            let actual = PathWithPosition::parse_str(input);
-            assert_eq!(
-                actual, expected,
-                "For positive case input str '{input}', got a parse mismatch"
-            );
-        }
-    }
-
-    #[test]
-    fn path_with_position_parsing_negative() {
-        for (input, row, column) in [
-            ("test_file.rs:a", None, None),
-            ("test_file.rs:a:b", None, None),
-            ("test_file.rs::", None, None),
-            ("test_file.rs::1", None, None),
-            ("test_file.rs:1::", Some(1), None),
-            ("test_file.rs::1:2", None, None),
-            ("test_file.rs:1::2", Some(1), None),
-            ("test_file.rs:1:2:3", Some(1), Some(2)),
-        ] {
-            let actual = PathWithPosition::parse_str(input);
-            assert_eq!(
-                actual,
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row,
-                    column,
-                },
-                "For negative case input str '{input}', got a parse mismatch"
-            );
-        }
     }
 
     // Trim off trailing `:`s for otherwise valid input.

--- a/crates/util/src/paths.rs
+++ b/crates/util/src/paths.rs
@@ -421,144 +421,183 @@ mod tests {
         );
     }
 
-    // Trim off trailing `:`s for otherwise valid input.
     #[test]
-    fn path_with_position_parsing_special() {
-        #[cfg(not(target_os = "windows"))]
-        let input_and_expected = [
-            (
-                "test_file.rs:",
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row: None,
-                    column: None,
-                },
-            ),
-            (
-                "test_file.rs:1:",
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row: Some(1),
-                    column: None,
-                },
-            ),
-            (
-                "crates/file_finder/src/file_finder.rs:1902:13:",
-                PathWithPosition {
-                    path: PathBuf::from("crates/file_finder/src/file_finder.rs"),
-                    row: Some(1902),
-                    column: Some(13),
-                },
-            ),
-        ];
+    fn path_with_position_parse_posix_path() {
+        // Trim off trailing `:`s for otherwise valid input.
+        assert_eq!(
+            PathWithPosition::parse_str("test_file:10:1:"),
+            PathWithPosition {
+                path: PathBuf::from("test_file"),
+                row: Some(10),
+                column: Some(1)
+            }
+        );
 
-        #[cfg(target_os = "windows")]
-        let input_and_expected = [
-            (
-                "test_file.rs:",
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row: None,
-                    column: None,
-                },
-            ),
-            (
-                "test_file.rs:1:",
-                PathWithPosition {
-                    path: PathBuf::from("test_file.rs"),
-                    row: Some(1),
-                    column: None,
-                },
-            ),
-            (
-                "\\\\?\\C:\\Users\\someone\\test_file.rs:1902:13:",
-                PathWithPosition {
-                    path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: Some(13),
-                },
-            ),
-            (
-                "\\\\?\\C:\\Users\\someone\\test_file.rs:1902:13:15:",
-                PathWithPosition {
-                    path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: Some(13),
-                },
-            ),
-            (
-                "\\\\?\\C:\\Users\\someone\\test_file.rs:1902:::15:",
-                PathWithPosition {
-                    path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: None,
-                },
-            ),
-            (
-                "\\\\?\\C:\\Users\\someone\\test_file.rs(1902,13):",
-                PathWithPosition {
-                    path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: Some(13),
-                },
-            ),
-            (
-                "\\\\?\\C:\\Users\\someone\\test_file.rs(1902):",
-                PathWithPosition {
-                    path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: None,
-                },
-            ),
-            (
-                "C:\\Users\\someone\\test_file.rs:1902:13:",
-                PathWithPosition {
-                    path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: Some(13),
-                },
-            ),
-            (
-                "crates/utils/paths.rs",
-                PathWithPosition {
-                    path: PathBuf::from("crates\\utils\\paths.rs"),
-                    row: None,
-                    column: None,
-                },
-            ),
-            (
-                "C:\\Users\\someone\\test_file.rs(1902,13):",
-                PathWithPosition {
-                    path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: Some(13),
-                },
-            ),
-            (
-                "C:\\Users\\someone\\test_file.rs(1902):",
-                PathWithPosition {
-                    path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
-                    row: Some(1902),
-                    column: None,
-                },
-            ),
-            (
-                "crates/utils/paths.rs:101",
-                PathWithPosition {
-                    path: PathBuf::from("crates\\utils\\paths.rs"),
-                    row: Some(101),
-                    column: None,
-                },
-            ),
-        ];
+        assert_eq!(
+            PathWithPosition::parse_str("test_file.rs:"),
+            PathWithPosition {
+                path: PathBuf::from("test_file.rs"),
+                row: None,
+                column: None
+            }
+        );
 
-        for (input, expected) in input_and_expected {
-            let actual = PathWithPosition::parse_str(input);
-            assert_eq!(
-                actual, expected,
-                "For special case input str '{input}', got a parse mismatch"
-            );
-        }
+        assert_eq!(
+            PathWithPosition::parse_str("test_file.rs:1:"),
+            PathWithPosition {
+                path: PathBuf::from("test_file.rs"),
+                row: Some(1),
+                column: None
+            }
+        );
+    }
+
+    #[test]
+    #[cfg(not(target_os = "windows"))]
+    fn path_with_position_parse_posix_path_with_suffix() {
+        assert_eq!(
+            PathWithPosition::parse_str("crates/file_finder/src/file_finder.rs:1902:13:"),
+            PathWithPosition {
+                path: PathBuf::from("crates/file_finder/src/file_finder.rs"),
+                row: Some(1902),
+                column: Some(13),
+            }
+        );
+    }
+
+    #[test]
+    #[cfg(target_os = "windows")]
+    fn path_with_position_parse_windows_path() {
+        assert_eq!(
+            PathWithPosition::parse_str("crates\\utils\\paths.rs"),
+            PathWithPosition {
+                path: PathBuf::from("crates\\utils\\paths.rs"),
+                row: None,
+                column: None
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("C:\\Users\\someone\\test_file.rs"),
+            PathWithPosition {
+                path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
+                row: None,
+                column: None
+            }
+        );
+    }
+
+    #[test]
+    #[cfg(target_os = "windows")]
+    fn path_with_position_parse_windows_path_with_suffix() {
+        assert_eq!(
+            PathWithPosition::parse_str("crates\\utils\\paths.rs:101"),
+            PathWithPosition {
+                path: PathBuf::from("crates\\utils\\paths.rs"),
+                row: Some(101),
+                column: None
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs:1:20"),
+            PathWithPosition {
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
+                row: Some(1),
+                column: Some(20)
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("C:\\Users\\someone\\test_file.rs(1902,13)"),
+            PathWithPosition {
+                path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: Some(13)
+            }
+        );
+
+        // Trim off trailing `:`s for otherwise valid input.
+        assert_eq!(
+            PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs:1902:13:"),
+            PathWithPosition {
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: Some(13)
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs:1902:13:15:"),
+            PathWithPosition {
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: Some(13)
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs:1902:::15:"),
+            PathWithPosition {
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: None
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs(1902,13):"),
+            PathWithPosition {
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: Some(13),
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs(1902):"),
+            PathWithPosition {
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: None,
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("C:\\Users\\someone\\test_file.rs:1902:13:"),
+            PathWithPosition {
+                path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: Some(13),
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("C:\\Users\\someone\\test_file.rs(1902,13):"),
+            PathWithPosition {
+                path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: Some(13),
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("C:\\Users\\someone\\test_file.rs(1902):"),
+            PathWithPosition {
+                path: PathBuf::from("C:\\Users\\someone\\test_file.rs"),
+                row: Some(1902),
+                column: None,
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("crates/utils/paths.rs:101"),
+            PathWithPosition {
+                path: PathBuf::from("crates\\utils\\paths.rs"),
+                row: Some(101),
+                column: None,
+            }
+        );
     }
 
     #[test]

--- a/crates/util/src/paths.rs
+++ b/crates/util/src/paths.rs
@@ -99,10 +99,6 @@ impl<T: AsRef<Path>> PathExt for T {
 /// A delimiter to use in `path_query:row_number:column_number` strings parsing.
 pub const FILE_ROW_COLUMN_DELIMITER: char = ':';
 
-/// Extracts filename and row-column suffixes.
-/// Parenthesis format is used by [MSBuild](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-diagnostic-format-for-tasks) compatible tools
-// NOTE: All cases need to have exactly three capture groups for extract(): file_name, row and column.
-// Valid patterns that don't contain row and/or column should have empty groups in their place.
 const ROW_COL_CAPTURE_REGEX: &str = r"(?x)
     ([^\(]+)(?:
         \((\d+),(\d+)\) # filename(row,column)
@@ -110,12 +106,12 @@ const ROW_COL_CAPTURE_REGEX: &str = r"(?x)
         \((\d+)\)()     # filename(row)
     )
     |
-    ([^\:]+)(?:
-        \:(\d+)\:(\d+)  # filename:row:column
+    (.+?)(?:
+        \:+(\d+)\:(\d+)\:*$  # filename:row:column
         |
-        \:(\d+)()       # filename:row
+        \:+(\d+)\:*()$       # filename:row
         |
-        \:()()          # filename:
+        \:*()()$             # filename:
     )";
 
 /// A representation of a path-like string with optional row and column numbers.
@@ -139,14 +135,28 @@ impl PathWithPosition {
     }
 
     /// Parses a string that possibly has `:row:column` or `(row, column)` suffix.
+    /// Parenthesis format is used by [MSBuild](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-diagnostic-format-for-tasks) compatible tools
     /// Ignores trailing `:`s, so `test.rs:22:` is parsed as `test.rs:22`.
     /// If the suffix parsing fails, the whole string is parsed as a path.
+    ///
+    /// Be mindful that `test_file:10:1:` is a valid posix filename.
+    /// `PathWithPosition` class assumes that the ending position-like suffix is **not** part of the filename.
     ///
     /// # Examples
     ///
     /// ```
     /// # use util::paths::PathWithPosition;
     /// # use std::path::PathBuf;
+    /// assert_eq!(PathWithPosition::parse_str("test_file"), PathWithPosition {
+    ///     path: PathBuf::from("test_file"),
+    ///     row: None,
+    ///     column: None,
+    /// });
+    /// assert_eq!(PathWithPosition::parse_str("test_file:10"), PathWithPosition {
+    ///     path: PathBuf::from("test_file"),
+    ///     row: Some(10),
+    ///     column: None,
+    /// });
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs"), PathWithPosition {
     ///     path: PathBuf::from("test_file.rs"),
     ///     row: None,
@@ -169,12 +179,12 @@ impl PathWithPosition {
     /// # use util::paths::PathWithPosition;
     /// # use std::path::PathBuf;
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs:a"), PathWithPosition {
-    ///     path: PathBuf::from("test_file.rs"),
+    ///     path: PathBuf::from("test_file.rs:a"),
     ///     row: None,
     ///     column: None,
     /// });
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs:a:b"), PathWithPosition {
-    ///     path: PathBuf::from("test_file.rs"),
+    ///     path: PathBuf::from("test_file.rs:a:b"),
     ///     row: None,
     ///     column: None,
     /// });
@@ -185,7 +195,7 @@ impl PathWithPosition {
     /// });
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs::1"), PathWithPosition {
     ///     path: PathBuf::from("test_file.rs"),
-    ///     row: None,
+    ///     row: Some(1),
     ///     column: None,
     /// });
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1::"), PathWithPosition {
@@ -195,18 +205,18 @@ impl PathWithPosition {
     /// });
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs::1:2"), PathWithPosition {
     ///     path: PathBuf::from("test_file.rs"),
-    ///     row: None,
-    ///     column: None,
+    ///     row: Some(1),
+    ///     column: Some(2),
     /// });
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1::2"), PathWithPosition {
-    ///     path: PathBuf::from("test_file.rs"),
-    ///     row: Some(1),
+    ///     path: PathBuf::from("test_file.rs:1"),
+    ///     row: Some(2),
     ///     column: None,
     /// });
     /// assert_eq!(PathWithPosition::parse_str("test_file.rs:1:2:3"), PathWithPosition {
-    ///     path: PathBuf::from("test_file.rs"),
-    ///     row: Some(1),
-    ///     column: Some(2),
+    ///     path: PathBuf::from("test_file.rs:1"),
+    ///     row: Some(2),
+    ///     column: Some(3),
     /// });
     /// ```
     pub fn parse_str(s: &str) -> Self {
@@ -423,6 +433,35 @@ mod tests {
 
     #[test]
     fn path_with_position_parse_posix_path() {
+        // Test POSIX filename edge cases
+        // Read more at https://en.wikipedia.org/wiki/Filename
+        assert_eq!(
+            PathWithPosition::parse_str(" test_file"),
+            PathWithPosition {
+                path: PathBuf::from("test_file"),
+                row: None,
+                column: None
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("a:bc:.zip:1"),
+            PathWithPosition {
+                path: PathBuf::from("a:bc:.zip"),
+                row: Some(1),
+                column: None
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("one.second.zip:1"),
+            PathWithPosition {
+                path: PathBuf::from("one.second.zip"),
+                row: Some(1),
+                column: None
+            }
+        );
+
         // Trim off trailing `:`s for otherwise valid input.
         assert_eq!(
             PathWithPosition::parse_str("test_file:10:1:"),
@@ -456,11 +495,33 @@ mod tests {
     #[cfg(not(target_os = "windows"))]
     fn path_with_position_parse_posix_path_with_suffix() {
         assert_eq!(
+            PathWithPosition::parse_str("app-editors:zed-0.143.6:20240710-201212.log:34:"),
+            PathWithPosition {
+                path: PathBuf::from("app-editors:zed-0.143.6:20240710-201212.log"),
+                row: Some(34),
+                column: None,
+            }
+        );
+    }
+
+    #[test]
+    #[cfg(not(target_os = "windows"))]
+    fn path_with_position_parse_posix_path_with_suffix() {
+        assert_eq!(
             PathWithPosition::parse_str("crates/file_finder/src/file_finder.rs:1902:13:"),
             PathWithPosition {
                 path: PathBuf::from("crates/file_finder/src/file_finder.rs"),
                 row: Some(1902),
                 column: Some(13),
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("crate/utils/src/test:today.log:34"),
+            PathWithPosition {
+                path: PathBuf::from("crate/utils/src/test:today.log"),
+                row: Some(34),
+                column: None,
             }
         );
     }
@@ -530,17 +591,17 @@ mod tests {
         assert_eq!(
             PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs:1902:13:15:"),
             PathWithPosition {
-                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
-                row: Some(1902),
-                column: Some(13)
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs:1902"),
+                row: Some(13),
+                column: Some(15)
             }
         );
 
         assert_eq!(
             PathWithPosition::parse_str("\\\\?\\C:\\Users\\someone\\test_file.rs:1902:::15:"),
             PathWithPosition {
-                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs"),
-                row: Some(1902),
+                path: PathBuf::from("\\\\?\\C:\\Users\\someone\\test_file.rs:1902"),
+                row: Some(15),
                 column: None
             }
         );

--- a/crates/util/src/paths.rs
+++ b/crates/util/src/paths.rs
@@ -174,7 +174,7 @@ impl PathWithPosition {
     /// });
     /// ```
     ///
-    /// # Expected parsing results when encounter ill-formated inputs.
+    /// # Expected parsing results when encounter ill-formatted inputs.
     /// ```
     /// # use util::paths::PathWithPosition;
     /// # use std::path::PathBuf;
@@ -502,11 +502,7 @@ mod tests {
                 column: None,
             }
         );
-    }
 
-    #[test]
-    #[cfg(not(target_os = "windows"))]
-    fn path_with_position_parse_posix_path_with_suffix() {
         assert_eq!(
             PathWithPosition::parse_str("crates/file_finder/src/file_finder.rs:1902:13:"),
             PathWithPosition {

